sdl_viewer/src/main.rs

```diff
@@ -260,38 +264,40 @@ impl NodeViewContainer {
         while let Ok((node_id, node_data)) = self.node_data_receiver.try_recv() {
             // Put loaded node into hash map.
             self.requested.remove(&node_id);
-            self.node_views
-                .insert(node_id, NodeView::new(program, node_data));
-            // TODO(sirver): Use a LRU Cache to throw nodes out that we haven't used in a while.
+            self.node_views.insert(node_id, NodeView::new(program, node_data));
         }
 
-        match self.node_views.entry(*node_id) {
-            Entry::Vacant(_) => {
-                // Limit the number of requested nodes because after a camera move
-                // requested nodes might not be in the frustum anymore.
-                if !self.requested.contains(&node_id) && self.requested.len() < 10 {  
-                    self.requested.insert(*node_id);
-                    self.node_id_sender.send(*node_id).unwrap();
-                }
-                None
-            }
-            Entry::Occupied(e) => Some(e.into_mut()),
+        if self.node_views.contains_key(node_id) {
+            return { self.node_views.get_mut(node_id).map(|f| f as &NodeView) }
         }
+        
+        // Limit the number of requested nodes because after a camera move
+        // requested nodes might not be in the frustum anymore.
+        if !self.requested.contains(&node_id) && self.requested.len() < 10 {  
+            self.requested.insert(*node_id);
+            self.node_id_sender.send(*node_id).unwrap();
+        }
+        None
     }
 
     fn request_all(&mut self, node_ids: &[octree::NodeId]) {
         for &node_id in node_ids {
-            match self.node_views.entry(node_id) {
-                Entry::Vacant(_) => {
-                    if !self.requested.contains(&node_id) {
-                        self.requested.insert(node_id);
-                        self.node_id_sender.send(node_id).unwrap();
-                    }
+            if !self.node_views.contains_key(&node_id) {
+                if !self.requested.contains(&node_id) {
+                    self.requested.insert(node_id);
+                    self.node_id_sender.send(node_id).unwrap();
                 }
-                Entry::Occupied(_) => {},
             }
         }
     }
+
+    fn get_used_memory_bytes(&self) -> usize {
+        let mut used_memory_bytes = 0;
+        for (_node_id, node_view) in self.node_views.iter() {
```

Comment:
shorter and equivalent: `for (_, node_view) in &self.node_views {`

Reply:
or more functional:

~~~
self.node_views().iter().map(|_, node_view| node_view.used_memory_bytes).sum()
~~~